community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/executionplan/addEagernessIfNecessary.scala

```diff
@@ -47,42 +36,84 @@ object addEagernessIfNecessary extends (Pipe => Pipe) {
     toPipe.dup(sources.toList)
   }
 
-  private def nodesWriteInterference(from: Effects, to: Effects) = {
-    val fromWrites = from.effectsSet.collect {
-      case writes: WritesNodes => writes
-    }
+  private def wouldInterfere(from: Effects, to: Effects): Boolean = {
+//    val nodesInterfere = nodesReadInterference(from, to)
+//    val relsInterfere = from.contains(ReadsRelationships) && to.contains(WritesRelationships)
+
+    // NOTE: Leaf effects will not be considered unless effects have first been flattened with Effects.regardlessOfLeafEffects
+    val fromWithoutLeafInfo = from.regardlessOfLeafEffects
+    val toWithoutLeafInfo = to.regardlessOfLeafEffects
+
```

Comment:
Please don't commit the commented out code. We've got version control for this! :)